src/MSALWrapper/PCACache.cs

```diff
@@ -37,14 +39,21 @@ internal class PCACache
         /// </summary>
         /// <param name="logger">The logger.</param>
         /// <param name="tenantId">The tenant id.</param>
+        /// <param name="cacheFileName">The cache file name.</param>
         /// <param name="osxKeyChainSuffix">The osx key chain suffix.</param>
-        internal PCACache(ILogger logger, Guid tenantId, string osxKeyChainSuffix = null)
+        internal PCACache(ILogger logger, Guid tenantId, string cacheFileName, string osxKeyChainSuffix = null)
         {
             this.logger = logger;
             this.osxKeyChainSuffix = string.IsNullOrWhiteSpace(osxKeyChainSuffix) ? $"{tenantId}" : $"{osxKeyChainSuffix}.{tenantId}";
             string appData = Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData);
             this.cacheDir = Path.Combine(appData, ".IdentityService");
-            this.cacheFileName = $"msal_{tenantId}.cache";
+
+            if (string.IsNullOrWhiteSpace(cacheFileName))
+            {
+                throw new ArgumentNullException($"{nameof(cacheFileName)} should not be null or whitespace.");
+            }
+
+            this.cacheFileName = cacheFileName;

```

Comment:
It's unclear to me how MSAL uses the cache Dir if the env name is a different complete full PATH. We should amke sure we understand how this works. 

I think we might want to actually take the full path for a cache file - and here use Path methods to split the parent dir and file into their separate parts, since that's what we need to give to the cache storage options. 

Does my confusion / worry make sense?

Reply:
I'm guessing since we tested setting the fileName arg to a full path in our hotfix, that that actually works, but if this is the case, we should at least comment this awkwardness.

Reply:
After I read the interfaces from MSAL, I think it may become more complicated if we allow user to use absolute path.  The signature is 
```csharp
public StorageCreationPropertiesBuilder(string cacheFileName, string cacheDirectory)
```
Is it worth to deal with the path in our side?

Reply:
We have to allow the user to give an absolute path (and in the internal version already do)